### PR TITLE
Strip Azure EventHub connection string in debug logs

### PR DESCRIPTION
**Pull request by exekias**

### CODE REVIEW

File: x-pack/filebeat/input/azureeventhub/input_test.go [hunk 1]
```diff
@@ -120,6 +120,30 @@ func TestNewInputDone(t *testing.T) {
 	inputtest.AssertNotStartedInputCanBeDone(t, NewInput, &config)
 }
 
+func TestStripConnectionString(t *testing.T) {
+	tests := []struct {
+		connectionString, expected string
+	}{
+		{
+			"Endpoint=sb://something",
+			"(redacted)",
```
Comment (jsoriano) [inline x-pack/filebeat/input/azureeventhub/input_test.go:129]: Why is this redacted and not `Endpoint=sb://something`?

File: x-pack/filebeat/input/azureeventhub/input.go [hunk 2]
```diff
@@ -235,3 +236,15 @@ func (a *azureInput) parseMultipleMessages(bMessage []byte) []string {
...
+// Strip connection string to remove sensitive information
+// A connection string should look like this:
+// Endpoint=sb://dummynamespace.servicebus.windows.net/;SharedAccessKeyName=DummyAccessKeyName;SharedAccessKey=5dOntTRytoC24opYThisAsit3is2B+OGY1US/fuL3ly=
+// This code will remove everything after ';' so key information is stripped
+func stripConnectionString(c string) string {
+	if parts := strings.SplitN(c, ";", 2); len(parts) == 2 {
+		return parts[0]
+	}
+
+	return "(redacted)"
```
Comment (jsoriano) [inline x-pack/filebeat/input/azureeventhub/input.go:249]: Why returning `(redacted)` if there is nothing after the endpoint? Would it be ok to return `c`?

**Review by jsoriano — APPROVED** [review]
Looks good, only wondering about the use of `(redacted)` instead of the endpoint if it doesn't have parameters.

File: x-pack/filebeat/input/azureeventhub/input.go (hunk 2 above)
Comment (exekias) [reply x-pack/filebeat/input/azureeventhub/input.go:249]: This is my thinking here, perhaps I should add a comment about it:

They only way to pass credentials here is putting them on the connection string (@narph please confirm if this is correct.), so if for any reason `;` is not part of it there is something wrong going on, so I stay on the safe side.

Comment (jsoriano) [reply x-pack/filebeat/input/azureeventhub/input.go:249]: Oh ok, makes sense :+1:

Comment (jsoriano) [reply x-pack/filebeat/input/azureeventhub/input.go:249]: And +1 to add a comment mentioning that the connection string may contain the credentials.

Comment (exekias) [reply x-pack/filebeat/input/azureeventhub/input.go:251]: done b3635a7

Comment (jsoriano) [reply x-pack/filebeat/input/azureeventhub/input.go:251]: Thanks :+1:

**Review by jsoriano — APPROVED** [review]

**Pull request merged**